diff --git a/base_external/patches/linux/am335x-bone-common.dtsi b/base_external/patches/linux/am335x-bone-common.dtsi
index ee28b86..b6c1247 100644
--- a/base_external/patches/linux/am335x-bone-common.dtsi
+++ b/base_external/patches/linux/am335x-bone-common.dtsi
@@ -187,6 +187,20 @@
 		>;
 	};
    
+   dcan0_pins: pinmux_dcan0_pins {
+		pinctrl-single,pins = <
+			AM33XX_IOPAD(0x97c, PIN_INPUT  | MUX_MODE2) /* (D17) uart1_rtsn.dcan0_rx */
+			AM33XX_IOPAD(0x978, PIN_OUTPUT | MUX_MODE2) /* (D18) uart1_ctsn.dcan0_tx */
+		>;
+	};
+
+   dcan1_pins: pinmux_dcan1_pins {
+		pinctrl-single,pins = <
+			AM33XX_IOPAD(0x984, PIN_INPUT  | MUX_MODE2) /* (D15) uart1_txd.dcan1_rx */
+			AM33XX_IOPAD(0x980, PIN_OUTPUT | MUX_MODE2) /* (D16) uart1_rxd.dcan1_tx */
+		>;
+	};
+   
 };
 
 &uart0 {
@@ -253,7 +267,7 @@
 
 &i2c2 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&i2c2_pins>;
+	//pinctrl-0 = <&i2c2_pins>;
 
 	status = "okay";
 	clock-frequency = <100000>;
@@ -299,6 +313,20 @@
 	};
 };
 
+&dcan0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&dcan0_pins>;
+
+	status = "okay";
+};
+
+&dcan1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&dcan1_pins>;
+
+	status = "okay";
+};
+
 /include/ "tps65217.dtsi"
 
 &tps {
